docs/tutorials/clustering/kclust.ipynb

```diff
@@ -0,0 +1,183 @@
+    "\n",
+    "where $N$ is the number of samples, $\\mu_{c}$ is the mean of cluster $c$, $C$ is the set of all clusters, $x_{i}$ is a sample point, and $|| \\ \\ \\ \\ \\ ||$ is the Euclidean distance.  \n",
+    "$$  \n",
+    "$$  \n",
+    "Let's look at an example using synthetic data\n",
+    "\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from graspologic.cluster.kclust import KMeansCluster\n",
+    "\n",
+    "# Synthetic data\n",
+    "\n",
+    "# Dim 1\n",
+    "class_1 = np.random.randn(150, 1)\n",
+    "class_2 = 2 + np.random.randn(150, 1)\n",
+    "dim_1 = np.vstack((class_1, class_2))\n",
+    "\n",
+    "# Dim 2\n",
+    "class_3 = np.random.randn(150, 1)\n",
+    "class_4 = 2 + np.random.randn(150, 1)\n",
+    "dim_2 = np.vstack((class_3, class_4))\n",
+    "\n",
+    "X = np.hstack((dim_1, dim_2))\n",
+    "\n",
+    "# Labels\n",
+    "label_1 = np.zeros((150, 1))\n",
+    "label_2 = 1 + label_1\n",
+    "\n",
+    "c = np.vstack((label_1, label_2)).reshape(300,)\n",
+    "\n",
+    "# Fit model\n",
+    "kclust_ = KMeansCluster(max_clusters=2)\n",
+    "\n",
+    "c_hat_kclust = kclust_.fit_predict(X,c)"
+   ]
+  },
+  {
+   "source": [
+    "We can compare our method to the existing implementation of KMeans clustering in Sklearn. Our method expands upon the existing Sklearn framework by allowing the user to automatically find the optimal number of clusters and achieve the best clustering possible. We can compare each optimal kclust model with the default KMeans model SKlearn generates."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.cluster import KMeans\n",
+    "from sklearn.metrics import confusion_matrix\n",
+    "from scipy.optimize import linear_sum_assignment\n",
+    "\n",
+    "# Default Sklearn KMeans\n",
+    "k_default_ = KMeans()\n",
+    "\n",
+    "c_hat_k_default = k_default_.fit_predict(X)\n",
+    "\n",
+    "# Function to remap labels for direct comparison\n",
+    "def remap_labels(y_true, y_pred):\n",
+    "\n",
+    "    confusion_mat = confusion_matrix(y_true, y_pred)\n",
+    "    row_inds, col_inds = linear_sum_assignment(confusion_mat, maximize=True)\n",
+    "    label_map = dict(zip(col_inds, row_inds))\n",
+    "    remapped_y_pred = np.vectorize(label_map.get)(y_pred)\n",
+    "    \n",
+    "    return remapped_y_pred\n",
+    "\n",
+    "# Remap Predicted labels\n",
+    "c_hat_kclust = remap_labels(c, c_hat_kclust)\n",
+    "c_hat_k_default = remap_labels(c, c_hat_k_default)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plotting Synthetic clusters\n",
```

Comment:
Same comment as autogmm about two consecutive code blocks.  Maybe make one output the automatically generated optimal cluster number?